cpp/src/arrow/filesystem/gcsfs.cc

```diff
@@ -233,6 +244,7 @@ class GcsOutputStream : public arrow::io::OutputStream {
 
  private:
   gcs::ObjectWriteStream stream_;
+  const io::IOContext io_context_;
```

Comment:
I don't know if this matters now, but since the context might have indirect references to `stream_`, it's better to put it before.

Reply:
> since the context might have indirect references to `stream_`

When would this happens?

Reply:
It's a weak reference, not an owning reference. The goal is to not destroy the `stream_` before the context is destroyed.